### PR TITLE
Don't need root on macOS

### PR DESCRIPTION
**Pull request by krlmlr**
At least not on my machine.

Includes a change to make the `build/` directory very explicit.

### CODE REVIEW

**Comment by krlmlr** [comment]
Thanks, I reduced the PR to the part I really care about.

File: docs/developers/developers-setup.md
Comment (MightyCreak) [inline docs/developers/developers-setup.md:171]: It doesn't require admin priviledges all the time?

File: docs/developers/developers-setup.md
Comment (MightyCreak) [inline docs/developers/developers-setup.md:168]: I'm actually wondering, as I can't test on macOS, is the app installed in `/usr/local/`? (that's the case for Linux in most distributions, but I'm unsure for macOS)

Comment (krlmlr) [reply docs/developers/developers-setup.md:168]:
```sh
~ which diffuse
/opt/homebrew/bin/diffuse
```

And there is a shortcut in the list of applications, but I'm not sure where it lives in the file system.

File: docs/developers/developers-setup.md
Comment (krlmlr) [reply docs/developers/developers-setup.md:171]: Worked without, never asked for a password.

Comment (MightyCreak) [reply docs/developers/developers-setup.md:171]: Would be nice to know if it never requires it, or if it's just because you have special settings on your system, but for now, it'll do just fine.

File: docs/developers/developers-setup.md
Comment (MightyCreak) [reply docs/developers/developers-setup.md:168]: Could you change `/usr/local/` with `/opt/homebrew/`? So it's done as well?

Comment (krlmlr) [reply docs/developers/developers-setup.md:168]: I believe this depends on the architecture, arm64 vs. Intel. We can mention both.

Comment (MightyCreak) [reply docs/developers/developers-setup.md:168]: wait... isn't `/opt/homebrew/` the directory if you installed Diffuse using `brew`? I don't know much about Mac, but I would find it weird that `meson install` would actually install it in a brew-specific directory, don't you think?

Comment (hugoholgersson) [reply docs/developers/developers-setup.md:168]: Yep, `meson install` will install Diffuse into /Applications and to /usr/local/bin, by default, in Mac. 

I wrote https://github.com/MightyCreak/diffuse/pull/165#discussion_r1005489000 about that as well. 

Maybe you would you like to fix https://github.com/MightyCreak/diffuse/pull/165#discussion_r1005489000 and https://github.com/MightyCreak/diffuse/pull/165#discussion_r1005739383 in this PR?

**Review by hugoholgersson — CHANGES_REQUESTED** [review]

File: docs/developers/developers-setup.md
Comment (krlmlr) [reply docs/developers/developers-setup.md:168]: Right, `/Applications` worked without `sudo` because I'm a member of the "admin" group on my system.

I'll do another pass here.

Regarding `/usr/local` vs. `/opt/homebrew`, I'm seeing this, with fairly default settings:

```
➜  build git:(b-macos-next-diff) meson install
ninja: Entering directory `/Users/kirill/git/diffuse/build'
ninja: no work to do.
Installing subdir /Users/kirill/git/diffuse/data/usr/share to /opt/homebrew/share
Installing /Users/kirill/git/diffuse/data/usr/share/omf/diffuse/diffuse-C.omf to /opt/homebrew/share/omf/diffuse
Installing /Users/kirill/git/diffuse/data/usr/share/omf/diffuse/diffuse-ru.omf to /opt/homebrew/share/omf/diffuse
Installing /Users/kirill/git/diffuse/data/usr/share/omf/diffuse/diffuse-it.omf to /opt/homebrew/share/omf/diffuse
Installing /Users/kirill/git/diffuse/data/usr/share/omf/diffuse/diffuse-cs.omf to /opt/homebrew/share/omf/diffuse
Installing /Users/kirill/git/diffuse/data/usr/share/man/it/man1/diffuse.1 to /opt/homebrew/share/man/it/man1
Installing /Users/kirill/git/diffuse/data/usr/share/man/cs/man1/diffuse.1 to /opt/homebrew/share/man/cs/man1
Installing /Users/kirill/git/diffuse/data/usr/share/man/ru/man1/diffuse.1 to /opt/homebrew/share/man/ru/man1
Installing /Users/kirill/git/diffuse/data/usr/share/man/man1/diffuse.1 to /opt/homebrew/share/man/man1
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/csharp.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/vb.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/bash.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/makefile.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/verilog.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/python.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/javascript.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/patch.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/octave.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/css.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/ini.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/pascal.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/jsp.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/html.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/c++.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/php.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/sql.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/xml.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/fortran.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/csh.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/tcl.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/gettext.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/objective-c++.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/rust.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/opencl.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/json.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/ruby.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/r.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/java.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/glsl.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/vhdl.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/perl.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/diffuse/syntax/erlang.syntax to /opt/homebrew/share/diffuse/syntax
Installing /Users/kirill/git/diffuse/data/usr/share/gnome/help/diffuse/it/diffuse.xml to /opt/homebrew/share/gnome/help/diffuse/it
Installing /Users/kirill/git/diffuse/data/usr/share/gnome/help/diffuse/cs/diffuse.xml to /opt/homebrew/share/gnome/help/diffuse/cs
Installing /Users/kirill/git/diffuse/data/usr/share/gnome/help/diffuse/ru/diffuse.xml to /opt/homebrew/share/gnome/help/diffuse/ru
Installing /Users/kirill/git/diffuse/data/usr/share/gnome/help/diffuse/C/diffuse.xml to /opt/homebrew/share/gnome/help/diffuse/C
Installing subdir /Users/kirill/git/diffuse/src/diffuse/vcs to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/vcs_registry.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/git.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/svk.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/rcs.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/svn.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/bzr.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/mtn.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/__init__.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/folder_set.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/darcs.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/cvs.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/hg.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing /Users/kirill/git/diffuse/src/diffuse/vcs/vcs_interface.py to /opt/homebrew/share/diffuse/diffuse/vcs
Installing po/cs/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/cs/LC_MESSAGES
Installing po/de/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/de/LC_MESSAGES
Installing po/es/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/es/LC_MESSAGES
Installing po/fr/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/fr/LC_MESSAGES
Installing po/it/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/it/LC_MESSAGES
Installing po/ja/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/ja/LC_MESSAGES
Installing po/ko/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/ko/LC_MESSAGES
Installing po/pl/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/pl/LC_MESSAGES
Installing po/pt/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/pt/LC_MESSAGES
Installing po/pt_BR/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/pt_BR/LC_MESSAGES
Installing po/ru/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/ru/LC_MESSAGES
Installing po/sv/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/sv/LC_MESSAGES
Installing po/th/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/th/LC_MESSAGES
Installing po/zh_CN/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/zh_CN/LC_MESSAGES
Installing po/zh_TW/LC_MESSAGES/diffuse.mo to /opt/homebrew/share/locale/zh_TW/LC_MESSAGES
Installing /Users/kirill/git/diffuse/build/data/diffuserc to /opt/homebrew/etc
Installing /Users/kirill/git/diffuse/data/icons/hicolor/scalable/apps/io.github.mightycreak.Diffuse.svg to /opt/homebrew/share/icons/hicolor/scalable/apps
Installing /Users/kirill/git/diffuse/data/icons/hicolor/symbolic/apps/io.github.mightycreak.Diffuse-symbolic.svg to /opt/homebrew/share/icons/hicolor/symbolic/apps
Installing /Users/kirill/git/diffuse/data/icons/diffuse.icns to /Applications/Diffuse.app/Contents/Resources
Installing /Users/kirill/git/diffuse/build/src/diffuse/diffuse_impl to /opt/homebrew/bin
Installing /Users/kirill/git/diffuse/build/src/diffuse/diffuse to /opt/homebrew/bin
Installing /Users/kirill/git/diffuse/build/src/diffuse/mac_launcher.sh to /Applications/Diffuse.app/Contents/MacOS
Installing /Users/kirill/git/diffuse/build/src/diffuse/Info.plist to /Applications/Diffuse.app/Contents
Installing /Users/kirill/git/diffuse/src/diffuse/__init__.py to /opt/homebrew/share/diffuse/diffuse
Installing /Users/kirill/git/diffuse/src/diffuse/constants.py to /opt/homebrew/share/diffuse/diffuse
Installing /Users/kirill/git/diffuse/src/diffuse/dialogs.py to /opt/homebrew/share/diffuse/diffuse
Installing /Users/kirill/git/diffuse/src/diffuse/main.py to /opt/homebrew/share/diffuse/diffuse
Installing /Users/kirill/git/diffuse/src/diffuse/preferences.py to /opt/homebrew/share/diffuse/diffuse
Installing /Users/kirill/git/diffuse/src/diffuse/resources.py to /opt/homebrew/share/diffuse/diffuse
Installing /Users/kirill/git/diffuse/src/diffuse/utils.py to /opt/homebrew/share/diffuse/diffuse
Installing /Users/kirill/git/diffuse/src/diffuse/widgets.py to /opt/homebrew/share/diffuse/diffuse
```

Comment (hugoholgersson) [reply docs/developers/developers-setup.md]: Do you have /opt/homebrew in any if your env variables?
```
export | grep homebrew
```

Comment (krlmlr) [reply docs/developers/developers-setup.md]:
```
HOMEBREW_CELLAR=/opt/homebrew/Cellar
HOMEBREW_PREFIX=/opt/homebrew
HOMEBREW_REPOSITORY=/opt/homebrew
INFOPATH=/opt/homebrew/share/info:
MANPATH=/opt/homebrew/share/man::
PATH=/Users/kirill/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

File: docs/developers/developers-setup.md
Comment (hugoholgersson) [inline docs/developers/developers-setup.md:180]: Command line args can actualy be passed to `diffuse` too (otherwise `diffuse` would not had been compatible with `git difftool`) - it's just that `open` seems to swollow all output so you don't see it.

Maybe we can fix that? Running `diffuse_impl` directly is not ideal because [Diffuse will not get its icon, etc](
https://github.com/MightyCreak/diffuse/blob/master/src/diffuse/mac-os-app/diffuse-mac.in#L5). If we can find a fix for `diffuse` or `open`, that would be much better than recommending users to call the internal impl.

**Review by hugoholgersson — CHANGES_REQUESTED** [review]

File: docs/developers/developers-setup.md
Comment (hugoholgersson) [reply docs/developers/developers-setup.md]: Ok. Yeah, I don't have the env variable `HOMEBREW_PREFIX` set and if I ask brew for its config I get `HOMEBREW_PREFIX: /usr/local`.

```
$ brew config
HOMEBREW_VERSION: 3.6.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 99682c5af3b48e7b37bf9c565781800153ee88fb
Last commit: 4 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 20a991bff175a2e7e89faf9fd9f4456163bce42f
Core tap last commit: 3 weeks ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
...
```

I found this hint in the install script of brew:
https://github.com/Homebrew/install/blob/master/install.sh#L132

File: docs/developers/developers-setup.md
Comment (hugoholgersson) [inline docs/developers/developers-setup.md:168]: Maybe skip the trailing slash on `/usr/local/` too. Here and on line 111.

**Comment by MightyCreak** [comment]
Since I don't have a Mac, I'll let your iron out the details here before merging. It seems there are a few questions pending:
* Where exactly does the app is installed? (why does meson take the HOMEBREW_PREFIX env?)
* Is it possible to run Diffuse on the command-line without having to use `diffuse_impl`?

For the last point, could it be that a shell wrapper should be installed in something like /usr/local/bin and it would take the command-line arguments and pass it through the Mac application? (There must be a way to send arguments to a Mac application, right?)

File: docs/developers/developers-setup.md
Comment (MightyCreak) [inline docs/developers/developers-setup.md:151]: Why add "using Meson" here? There are no other build systems for macOS. I'd rather not add this information as it could lead to some confusion.

(I'm ok with "Mac OS" becoming "macOS", I wanted to change that, but since the first PR about Mac wrote Mac OS, I wanted to use the same terms)

File: docs/developers/developers-setup.md
Comment (MightyCreak) [inline docs/developers/developers-setup.md:171]: Since you wrote this:

> [It] worked without sudo because I'm a member of the "admin" group on my system

I think you can revert to "requires", because no matter if it asks the question or not, it does require the user to have the admin privileges.

File: docs/developers/developers-setup.md
Comment (krlmlr) [reply docs/developers/developers-setup.md:180]: I'll extract this to a separate PR so that we don't forget.

File: docs/developers/developers-setup.md
Comment (krlmlr) [reply docs/developers/developers-setup.md]: I'm just a user here, set it up with the install script and didn't touch anything. I believe it's fair to assume that other users will see the same behavior. No idea if it's correct though.

Comment (hugoholgersson) [reply docs/developers/developers-setup.md]: Yep yep, https://github.com/Homebrew/install/blob/master/install.sh#L132Y indicates that users of M1 CPUs will get this path?

**Review by hugoholgersson — APPROVED** [review]

**Comment by MightyCreak** [comment]
Fixed a few remaining changes that I didn't really agreed on + some markdown refactoring to comply with the linter. Will merge once pipeline is green

**Pull request merged**
